py3status/modules/wwan_status_nm.py

```diff
@@ -0,0 +1,283 @@
+                            status['access-technologies'].bit_length() - 1)
+                        data['netgen'] = self.speed[highest_access_bit]
+                    except:
+                        data['netgen'] = STRING_UNKNOWN
+
+                    # if registred on network, get operator name
+                    try:
+                        if self.rstates[status[
+                                'm3gpp-registration-state']] in ('HOME',
+                                                                 'ROAMING'):
+                            data['operator'] = status['m3gpp-operator-name']
+                        else:
+                            data['operator'] = STRING_UNKNOWN
+                    except:
+                        data['operator'] = STRING_UNKNOWN
+
+                    # use human readable format
+                    data['status'] = self.states[status['state']]
+
+                    # TODO: maybe change to a rstates condition?
+                    if status['state'] >= 0 and status['state'] <= 7:  # no service
+                        color = self.py3.COLOR_DEGRADED
+                        full_text = self.py3.safe_format(
+                            self.format_no_service, data)
+
+                    elif status['state'] == 11:  # service
+                        # Get network config
+                        bearer = modem_proxy.Bearers[0]
+                        network_config = self._get_network_config(bearer)
+
+                        ipv4 = network_config.get('ipv4', {})
+                        ipv6 = network_config.get('ipv6', {})
+
+                        # Add network config to data dict
+                        data['format_ipv4'] = self.py3.safe_format(
+                            self.format_ipv4, ipv4)
+                        data['format_ipv6'] = self.py3.safe_format(
+                            self.format_ipv6, ipv6)
+
+                        if ipv4 or ipv6:
+                            if self.consider_3G_degraded and data['netgen'] != 'LTE':
+                                color = self.py3.COLOR_DEGRADED
+                            else:
+                                color = self.py3.COLOR_GOOD
+                            full_text = self.py3.safe_format(
+                                self.format_up, data)
+                        else:
+                            color = self.py3.COLOR_DEGRADED
+                            full_text = self.py3.safe_format(
+                                self.format_no_service, data)
+
+                    else:  # disconnected
+                        full_text = self.py3.safe_format(
+                            self.format_down, data)
+                        color = self.py3.COLOR_BAD
+
+                except:
+                    data['error'] = self.states[status['state']]
+                    color = self.py3.COLOR_BAD
+                    full_text = self.py3.safe_format(self.format_error, data)
```

Comment:
Which code can trigger this `self.format_error`? I'm fairly certain we can live without this part eventually.

Reply:
I will check

Reply:
I confirm remove, try and this except and all this to work now

Reply:
Use something like `py3status --log-file /tmp/py3status-bottom-bar.log` in your i3 config and add `self.py3.log('-' * 20)` in few different places... then you can find out where the exception occurs. It could be a simple bad `dictname[keyerror]` or something else.

Or easy...Remove `try/except` and look at the error to understand this better. Share too.

Reply:
Ok thanks ;)